dev/connect-jvm-client-mima-check

```diff
@@ -34,20 +34,18 @@ fi
 
 rm -f .connect-mima-check-result
 
-echo "Build sql module, connect-client-jvm module and connect-client-jvm test jar..."
-build/sbt "sql/package;connect-client-jvm/assembly;connect-client-jvm/Test/package"
+echo "Build required modules..."
+build/sbt "sql/package;connect-client-jvm/assembly;connect-client-jvm/Test/package;avro/package"
 
 CONNECT_TEST_CLASSPATH="$(build/sbt -DcopyDependencies=false "export connect-client-jvm/Test/fullClasspath" | grep jar | tail -n1)"
-CONNECT_CLASSPATH="$(build/sbt -DcopyDependencies=false "export connect-client-jvm/fullClasspath" | grep jar | tail -n1)"
 SQL_CLASSPATH="$(build/sbt -DcopyDependencies=false "export sql/fullClasspath" | grep jar | tail -n1)"
 
-
 echo "Do connect-client-jvm module mima check ..."
 
 $JAVA_CMD \
   -Xmx2g \
   -XX:+IgnoreUnrecognizedVMOptions --add-opens=java.base/java.util.jar=ALL-UNNAMED \
-  -cp "$CONNECT_CLASSPATH:$CONNECT_TEST_CLASSPATH:$SQL_CLASSPATH" \
+  -cp "$CONNECT_TEST_CLASSPATH:$SQL_CLASSPATH" \
```

Comment:
@hvanhovell Yes, you are right. Only `checkDatasetApiCompatibility` needs `SQL_CLASSPATH ` because it is not a pure mima check.

So we just need `CONNECT_TEST_CLASSPATH` for `CheckConnectJvmClientCompatibility ` and `SQL_CLASSPATH` for `checkDatasetApiCompatibility`